threeML/plugins/spectrum/pha_spectrum.py

```diff
@@ -254,14 +257,23 @@ def _read_pha_or_pha2_file(pha_file_or_instance, spectrum_number=None, file_type
         key_has_been_collected = False
 
         if keyname in header:
-            gathered_keywords[internal_name] = header.get(keyname)
+            if keyname in _required_keyword_types and type(header.get(keyname)) is not _required_keyword_types[keyname]:
+                warnings.warn("unexpected type of %(keyname)s, expected %(expected_type)s\n found %(found_type)s: %(found_value)s"%
+                        dict(
+                                keyname=keyname,
+                                expected_type=_required_keyword_types[keyname],
+                                found_type=type(header.get(keyname)),
+                                found_value=header.get(keyname),
+                            ))
+            else:
+                gathered_keywords[internal_name] = header.get(keyname)
 
```

Comment:
I was just using it for INTEGRAL, so I pinpoint the examples. I will try to add more with time!